src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentUriExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Razor;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal static class DocumentUriExtensions
+    {
+        public static string GetAbsoluteOrUNCPath(this DocumentUri documentUri)
+        {
+            if (documentUri is null)
+            {
+                throw new ArgumentNullException(nameof(documentUri));
+            }
+
+            return documentUri.ToUri().GetAbsoluteOrUNCPath();
```

Comment:
I know this is still doing ToUri and creating additional objects, but getting the detection of UNC and absolute paths correct is harder than I think we want to deal with in this PR, to I used extension methods to wrap everything so we can deal with it later.

Reply:
ya that's fine

Reply:
DocumentUri should support unc and other paths, it's based off the vscode implementation with a suite of tests. This was important for @TylerLeonhardt .

Reply:
[This test suite was shamelessly pilfered from vscode](https://github.com/OmniSharp/csharp-language-server-protocol/blob/383d6c2509548fd1292cad339f8e209f4308b868/test/Lsp.Tests/VsCodeDocumentUriTests.cs).  [And there are some additional ones as well just ensuring it works with various different character types](https://github.com/OmniSharp/csharp-language-server-protocol/blob/383d6c2509548fd1292cad339f8e209f4308b868/test/Lsp.Tests/DocumentUriTests.cs).

If there is one thing I'm an expert of, I'm not an expert of anything (I'm not...) it's converting C# to TypeScript. or TypeScript to C#, lol.